config/e2e/metricbeat.yaml

```diff
@@ -0,0 +1,295 @@
+  name: metricbeat
+  namespace: {{ .E2ENamespace }}
+  labels:
+    k8s-app: metricbeat
+spec:
+  selector:
+    matchLabels:
+      k8s-app: metricbeat
+  template:
+    metadata:
+      labels:
+        k8s-app: metricbeat
+    spec:
+      serviceAccountName: metricbeat
+      terminationGracePeriodSeconds: 30
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
+      containers:
+        - name: metricbeat
+          image: docker.elastic.co/beats/metricbeat:7.7.0
+          args: [
+            "-c", "/etc/metricbeat.yml",
+            "-e",
+            "-system.hostfs=/hostfs",
+            "-d", "autodiscover",
+            "-d", "kubernetes",
+          ]
+          env:
+            - name: ELASTICSEARCH_HOST
+              valueFrom:
+                secretKeyRef:
+                  name: "eck-{{ .TestRun }}"
+                  key: monitoring-ip
+            - name: ELASTICSEARCH_PORT
+              value: "9200"
+            - name: ELASTICSEARCH_USERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: "eck-{{ .TestRun }}"
+                  key: monitoring-user
+            - name: ELASTICSEARCH_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: "eck-{{ .TestRun }}"
+                  key: monitoring-pass
+            - name: NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+          securityContext:
+            runAsUser: 0
+            {{ if .OcpCluster }}
+            privileged: true
+            {{ end }}
+          resources:
+            limits:
+              memory: 200Mi
+            requests:
+              cpu: 100m
+              memory: 100Mi
+          volumeMounts:
+            - name: config
+              mountPath: /etc/metricbeat.yml
+              readOnly: true
+              subPath: metricbeat.yml
+            - name: modules
+              mountPath: /usr/share/metricbeat/modules.d
+              readOnly: true
+            - name: dockersock
+              mountPath: /var/run/docker.sock
+            - name: proc
+              mountPath: /hostfs/proc
+              readOnly: true
+            - name: cgroup
+              mountPath: /hostfs/sys/fs/cgroup
+              readOnly: true
+            - name: monitoring-ca
+              mountPath: /mnt/elastic/monitoring-ca.crt
+              readOnly: true
+              subPath: monitoring-ca
+      volumes:
+        - name: proc
+          hostPath:
+            path: /proc
+        - name: cgroup
+          hostPath:
+            path: /sys/fs/cgroup
+        - name: dockersock
+          hostPath:
+            path: /var/run/docker.sock
+        - name: config
+          configMap:
+            defaultMode: 0600
+            name: metricbeat-daemonset-config
+        - name: modules
+          configMap:
+            defaultMode: 0600
+            name: metricbeat-daemonset-modules
+        - name: data
```

Comment:
This is something that's already wrong in our Metricbeat recipe: the data volume is not used in any mount afaik and if it were I believe the permissions would be incorrect for a non-root user.

Reply:
I'll fix this and raise a PR against the recipe.